src/app/core/store/documents/documents.reducer.ts

```diff
@@ -30,8 +30,18 @@ export function documentsReducer(
       return addDocuments(state, action);
     case DocumentsActionType.CREATE_SUCCESS:
       return addOrUpdateDocument(state, action.payload.document);
+    case DocumentsActionType.UPDATE_DATA_INTERNAL:
+      return updateDocument(state, action);
+    case DocumentsActionType.PATCH_DATA_INTERNAL:
+      return patchDocument(state, action);
     case DocumentsActionType.UPDATE_SUCCESS:
       return addOrUpdateDocument(state, action.payload.document);
+    case DocumentsActionType.UPDATE_FAILURE:
+      if (action.payload.originalDocument) {
+        return documentsAdapter.upsertOne(action.payload.originalDocument, state);
```

Comment:
We should not always update it but instead check the version and proceed only if the document in the store is not newer than the original document.